cpp/src/plasma/client.cc

```diff
@@ -514,7 +515,9 @@ Status PlasmaClient::Impl::GetBuffers(
         auto iter = gpu_object_map.find(object_ids[i]);
         ARROW_CHECK(iter != gpu_object_map.end());
         iter->second->client_count++;
-        physical_buf = iter->second->ptr;
+        physical_buf = std::make_shared<CudaBuffer>(iter->second->ptr->mutable_data(),
```

Comment:
It looks like this logic (creating a `CudaBuffer` from a `GpuProcessHandle`) is duplicated. Can you factor it out?

Reply:
thread 1:
{
// assume that obj_id1 is already contained

ObjectBuffer buff1;
client.Get(&obj_id1, 1, -1, &buff1);

ObjectBuffer buff2;
client.Get(&obj_id1, 1, -1, &buff2); // here will trigger the logic

// not to set buff1.data, buff2.data nullptr
client.Release(obj_id1);
client.Release(obj_id1);

client.Delete(obj_id1);
}
// let buff1, buff2 auto release here.

thread 2:
{
std::shared_ptr buff;
plasma_client2.Create(object_id2, size, nullptr, 0, &buff, 1);
// If the address allocated by the server is just the object_id1 released, error occur!
}

Reply:
@pitrou Do yo have any more question about this PR?

Reply:
@shengjun1985 I think you misunderstood my comments above.
Can you add some tests, and perhaps also factor out the function that creates a `CudaBuffer` from a `GpuProcessHandle`? Thank you.

Reply:
I find that the error will also occur in a signal thread.
I add a gtest and some comments now. Please help to review.

Reply:
Ok... I'll do the refactor for you.